### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by ktorpi**
Updated JerryScript Internals documentation form section High-Level design to section Virtual Machine.

JerryScript-DCO-1.0-Signed-off-by: István Kádár ikadar@inf.u-szeged.hu

### CODE REVIEW

File: 04.internals.md [hunk 1]
```diff
@@ -8,469 +8,211 @@ permalink: /internals/
...
+<div class="CSSTableGenerator" markdown="block">
 
-|op_with|
-||
-||
-|...|
-|op_meta (end_with)|
+| byte-code                            | description                                       |
+| CBC_PUSH_LITERAL                     | Pushes the value of the given literal argument.     |
+| CBC_PUSH_TWO_LITERALS                | Pushes the value of the given two literal argument. |
```
Comment (zherczeg) [inline 04.internals.md]: two literal arguments

File: 04.internals.md [hunk 2]
```diff
@@ -8,469 +8,211 @@ permalink: /internals/
...
+| CBC_PUSH_UNDEFINED                   | Pushes an undefined value.                          |
+| CBC_PUSH_TRUE                        | Pushes a logical true.                              |
+| CBC_PUSH_PROP_LITERAL                | Pushes a property whose base object is popped from the stack, and the property name is given in literal argument. |
```
Comment (zherczeg) [inline 04.internals.md]: property name is passed as the literal argument

File: 04.internals.md [hunk 3]
```diff
@@ -8,469 +8,211 @@ permalink: /internals/
...
-|...|
-|op_meta (catch)|
-|...|
-|op_meta (finally)|
-|...|
-|op_meta (end_try_catch_finally)|
+| byte-code                            | description |
+| CBC_CALL0                            | Calls a function without arguments. The return value won't be pushed onto the stack. |
+| CBC_CALL1                            | Calls a function with one argument. The return value won't be pushed onto the stack. |
+| CBC_CALL                             | Calls a function with n arguments. n is given in argument. The return value won't be pushed onto the stack. |
```
Comment (zherczeg) [inline 04.internals.md]: The n is passed as a byte argument.

File: 04.internals.md [hunk 4]
```diff
@@ -8,469 +8,211 @@ permalink: /internals/
...
-`prop_num` - number of properties
-
-<div class="CSSTableGeneratorBlock" markdown="block">
+<div class="CSSTableGenerator" markdown="block">
 
-|op_obj_decl|
-|... <br>(intermediate evaluation of value/function expression, etc.)|
-|op_meta (prop_data/ prop_getter/ prop_setter)|
+| byte-code                            | description |
+| CBC_LOGICAL_NOT                      | Negates the logical value that poped from the stack. The result is pushed onto the stack. |
```
Comment (zherczeg) [inline 04.internals.md]: I think it is popped not poped

File: 04.internals.md [hunk 5]
```diff
@@ -481,72 +223,62 @@ The major structure for data representation is `ECMA_value`. Lower two bits of t
...
-- is_stack_allocated - some temporary strings are stack_allocated to reduce loading of memory (perf)
-- container - type of actual string storage/encoding
-- hash - hash, calculated from two last characters (for faster comparison (perf))
-- literal identifier - actual string is in the literal storage
-- magic_string_id - string is equal to one of engine's magic strings
-- uint32 - string is represented with unsigned integers (useful for array indexing)
-- number_cp (compressed pointer to number) - string is represented with floating point number
-- collection_cp - string is stored in one or several pool's chunks (see also: chars collection, collection header, collection chunk)
-- concatenation_1_cp, concatenation_2_cp - pointers to two strings (parts of concatenation)
+### String
```
Comment (zherczeg) [inline 04.internals.md]: I think from this part followup updates are needed.

File: css/img.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+img[alt="Number"] {
+  max-width:  25%;
+  display: block;
+}
+
+img[alt="Compressed Pointer"] {
+  max-width:  25%;
+  display: block;
+}
```
Comment (zherczeg) [inline css/img.css]: No newline?

**Comment by LaszloLango** [comment]
LGTM

**Comment by zherczeg** [comment]
LGTM

**Pull request merged**